### ISSUE TITLE
Upgrade `@keyv/redis` and configure to not use a set to track keys

### ISSUE THREAD

**Issue by joselcvarela** [directus/directus#20205]
### Describe the Improvement

Sometimes user sees content cached when it was already updated.
The project is configured with `CACHE_AUTO_PURGE` to true and `CACHE_TTL` to 30m.
So the user only sees content updated after the TTL.

We already have tried to use the clear cache endpoint without any effect.

This is difficult to reproduce but I have seen this once in a while 
I believe `@keyv/redis` is the culprit. It looks like somehow it loses track of some keys and those are only deleted after the TTL set on Redis.

It looks like we are still using version 2.5.8 of `@keyv/redis` but the latest version is 2.8.0: https://github.com/directus/directus/blob/main/api/package.json#L222C15-L222C15

In this version there's a new feature where keys can be stored as simple keys instead of being tracked on a set (`useRedisSets: false`): https://github.com/jaredwray/keyv/pull/855/files
In the past I have noticed that in other projects, Redis instances were growing with keys not being deleted. So I think this would be nice to have.

Also the deletion of keys in most recent version is asynchronous by using UNLINK instead of DEL command.
https://github.com/jaredwray/keyv/pull/921/files

**Comment by ComfortablyCoding** [comment]
Hello, can i take this?

**Comment by rijkvanzanten** [comment]
_yes_ 👁️👄👁️

**Comment by ComfortablyCoding** [comment]
Based on  https://github.com/directus/directus/pull/20238#issuecomment-1797001648

Opened the following PR to add namespace support for `useRedisSets` false https://github.com/jaredwray/keyv/pull/956

Its been merged and stated to be released in the coming weeks.

This is blocked until then.

**Comment by alexchopin** [comment]
We need you thoughts on Directus KV @rijkvanzanten

**Comment by ComfortablyCoding** [comment]
This is blocked by https://github.com/jaredwray/keyv/pull/964
